build_system/clerk_driver.ml

```diff
@@ -157,366 +176,507 @@ type expected_output_descr = {
+        let dir' = File.dirname dir in
+        if dir' = dir then None else aux dir'
+    in
+    lazy (aux (Sys.getcwd ()))
+
+  let exec_dir: File.t =
+    (* Do not use Sys.executable_name, which may resolve symlinks: we want the original path.
+       (e.g. _build/install/default/bin/foo is a symlink) *)
+    Filename.dirname Sys.argv.(0)
+
+  let clerk_exe: File.t Lazy.t = lazy Sys.executable_name
+
+  let catala_exe: File.t Lazy.t = lazy (
+    let f = File.(exec_dir / "catala") in
+    if Sys.file_exists f then f else
+      match Lazy.force catala_project_root with
+      | Some root -> File.(root/"_build"/"default"/"compiler"/"catala.exe")
+      | None -> "catala" (* Dynamically resolved from PATH *)
+  )
+
+  let build_dir: File.t Lazy.t = lazy (
+    File.(Sys.getcwd () / "_build")
+  )
+  (* TODO: probably detect the project root and put this there instead *)
+
+  (** Locates the main [lib] directory containing the OCaml libs *)
+  let ocaml_libdir: File.t Lazy.t =
+    lazy
+      (try String.trim (File.process_out "opam" ["var"; "lib"])
+       with Failure _ -> (
+           try String.trim (File.process_out "ocamlc" ["-where"])
+           with Failure _ -> (
+               match File.(check_directory (exec_dir /../ "lib")) with
+               | Some d -> d
+               | None ->
+                 Message.raise_error
+                   "Could not locate the OCaml library directory, make sure OCaml or \
+                    opam is installed")))
+
+  (** Locates the directory containing the OCaml runtime to link to *)
+  let ocaml_runtime_dir: File.t Lazy.t =
+    lazy
+      (let d =
+         match Lazy.force catala_project_root with
+         | Some root ->
+           (* Relative dir when running from catala source *)
+           File.(
+             root
+             / "_build"
+             / "install"
+             / "default"
+             / "lib"
+             / "catala"
+             / "runtime_ocaml")
+         | None -> (
+             match
+               File.check_directory
+                 File.(exec_dir /../ "lib" / "catala" / "runtime_ocaml")
+             with
+             | Some d -> d
+             | None -> File.(Lazy.force ocaml_libdir / "catala" / "runtime"))
+         (* FIXME check this, not "runtime_ocaml" ?? *)
+       in
+       match File.check_directory d with
+       | Some dir ->
+         Message.emit_debug "Catala runtime libraries found at @{<bold>%s@}." dir;
+         dir
+       | None ->
+         Message.raise_error
+           "@[<hov>Could not locate the Catala runtime library.@ Make sure that \
+            either catala is correctly installed,@ or you are running from the \
+            root of a compiled source tree.@]")
+
+  let ocaml_link_flags: string list Lazy.t = lazy (
+    let link_libs =
       [
-        Var.catala_cmd;
-        Var.tested_file;
-        Lit redirect;
-        Var.expected_output;
-        Lit "2>&1";
-        Lit "|| true";
+        "biniou";
```

Comment:
I have a fix pending, `yojson.1.7` depended on `biniou` but it's no longer the case with `2.0` so I am making the version constraint stricter and removing this dep.